servicetalk-tcp-netty-internal/src/main/java/io/servicetalk/tcp/netty/internal/TcpServerBinder.java

```diff
@@ -122,7 +122,7 @@ protected void initChannel(Channel channel) {
                                     // subscribeOn is required to offload calls to connectionAcceptor#accept
                                     .subscribeOn(executionContext.executor()));
                 }
-                connectionSingle.whenOnError(cause -> {
+                connectionSingle.afterOnError(cause -> {
```

Comment:
Is `before` not appropriate here as we are sending an error and hence the channel will not be usable?

Reply:
I guess it shouldn't matter ... was just trying to keep behavior the same. I will make it explicit `before*` rather than having to deal with this ambiguity in the future.